mesa/experimental/continuous_space/continuous_space.py

```diff
@@ -0,0 +1,271 @@
+"""A Continuous Space class."""
+
+import warnings
+from collections.abc import Iterable
+from itertools import compress
+from random import Random
+
+import numpy as np
+from numpy.typing import ArrayLike
+from scipy.spatial.distance import cdist
```

Comment:
I am conflicted about adding a core dependency for one call, although it is scipy. Is there thoughts or plan to grow this to add more types of calculations  (e.g. non Euclidian)  or use scpiy in other spots?

Reply:
Its part of anaconda, the code is roughly 4 times faster than a numpy only version, and you can use other distances with it, so I think it is worth it, but it can be changed if others see this differently.

Reply:
My bias is always for faster so I say leave it in as core or add to recommended (scipy is a part of examples). (I am also tracking numpy, pandas, and tqdm as [core](https://github.com/projectmesa/mesa/blob/1ce3f37730a5798135daf488bef8f1a23e746192/pyproject.toml#L37)

I also proposed adding `**kwargs` to `calculate_distances`  below so user can exploit some of the other cdist capability